lib/verifier.js

```diff
@@ -467,9 +467,25 @@ export class CertificateVerifier {
         return this._failed(completionCallback, reason, e);
       }
 
+      this._checkExpiryDate(completionCallback);
+    }
+
+  }
+
+  _checkExpiryDate(completionCallback) {
+    this.statusCallback(Status.checkingExpiryDate);
+
+    if (!this.certificate.expires) {
+      // expiry date not set, nothing to do
       return this._succeed(completionCallback);
     }
 
+    var expiryDate = Date.parse(this.certificate.expires);
+
+    if (new Date() >= expiryDate) {
+      var reason = "This certificate has expired.";
+      return this._failed(completionCallback, reason, null);
+    }
```

Comment:
Thanks for reviewing my PR! I had this uneasy feeling about something and turns out I missed putting `return this._succeed(completionCallback);` at the end of the function here. Sorry for the trouble.